l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,278 @@
+            active: true,
+            validatorWeight: _validatorWeight,
+            validatorPubKey: _validatorPubKey,
+            validatorPoP: _validatorPoP,
+            attesterWeight: _attesterWeight,
+            attesterPubKey: _attesterPubKey
+        });
+    }
+
+    /// @notice Deactivates a node, preventing it from participating in committees.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be inactivated.
+    function deactivate(address _nodeOwner) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].active = false;
+    }
+
+    /// @notice Activates a previously inactive node, allowing it to participate in committees.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be activated.
+    function activate(address _nodeOwner) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].active = true;
+    }
+
+    /// @notice Removes a node from the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be removed.
+    function remove(address _nodeOwner) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+        // Remove from array by swapping the last element (gas-efficient, not preserving order).
+        nodeOwners[nodeOwnerIdx(_nodeOwner)] = nodeOwners[nodeOwners.length - 1];
+        nodeOwners.pop();
+        // Remove from mapping.
+        delete nodes[_nodeOwner];
+    }
+
+    /// @notice Changes the validator weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose validator weight will be changed.
+    /// @param _weight The new validator weight to assign to the node.
+    function changeValidatorWeight(address _nodeOwner, uint256 _weight) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].validatorWeight = _weight;
+    }
+
+    /// @notice Changes the attester weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose attester weight will be changed.
+    /// @param _weight The new attester weight to assign to the node.
+    function changeAttesterWeight(address _nodeOwner, uint256 _weight) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+        nodes[_nodeOwner].attesterWeight = _weight;
```

Comment:
Please always emit events on changed parameters here and in other functions

Reply:
Added here: https://github.com/matter-labs/era-contracts/pull/555/commits/92b0fcc04c6d51e20a9a0c2fc64cd518d9af2d36

For the committees events I didn't use params, to avoid the overhead of handling both storage and memory variables.